src/pudl/metadata/classes.py

```diff
@@ -929,18 +935,58 @@ def get_resource_ids(self) -> List[str]:
             ]
         )
 
-    def get_temporal_coverage(self) -> str:
+    def get_temporal_coverage(self, partitions: Dict = None) -> str:
         """Return a string describing the time span covered by the data source."""
-        if "years" in self.working_partitions:
-            return f"{min(self.working_partitions['years'])}-{max(self.working_partitions['years'])}"
-        elif "year_month" in self.working_partitions:
-            return f"through {self.working_partitions['year_month']}"
+        if partitions is None:
+            partitions = self.working_partitions
+        if "years" in partitions:
+            return f"{min(partitions['years'])}-{max(partitions['years'])}"
+        elif "year_month" in partitions:
+            return f"through {partitions['year_month']}"
         else:
             return ""
 
-    def to_rst(self) -> None:
+    def add_datastore_metadata(self) -> None:
+        """Get source file metadata from the datastore."""
+        dp_desc = Datastore(sandbox=False).get_datapackage_descriptor(self.name)
+        partitions = dp_desc.get_partitions()
+        if "year" in partitions:
+            partitions["years"] = partitions["year"]
+        elif "year_month" in partitions:
+            partitions["year_month"] = max(partitions["year_month"])
+        self.source_file_dict["source_years"] = self.get_temporal_coverage(partitions)
+        self.source_file_dict["download_size"] = dp_desc.get_download_size()
+
+    def to_rst(
+        self,
+        docs_dir: DirectoryPath,
+        source_resources: List,
+        extra_resources: List,
+        output_path: str = None,
+    ) -> None:
         """Output a representation of the data source in RST for documentation."""
-        pass
+        self.add_datastore_metadata()
+        template = _get_jinja_environment(docs_dir).get_template(
+            f"{self.name}_child.rst.jinja"
+        )
+        pdfs = (docs_dir / f"data_sources/{self.name}").glob("*.pdf")
+        instruction_files_absolute = [x for x in pdfs if x.is_file()]
+        # make these paths relative to docs/data_sources dir instead of absolute
+        instruction_files = [
+            "/".join(str(abs_path).split("/")[-2:])
+            for abs_path in instruction_files_absolute
+        ]
```

Comment:
I think using `pathlib.Path` for this manipulation could be more readable, less platform dependent, and more robust against nonsense like filenames that contain the `/` character. I think it would look something like
```py
data_source_dir = docs_dir / "data_sources"
download_paths = [
    path.relative_to(data_source_dir)
    for path in (data_source_dir / self.name).glob("*.pdf")
    if path.is_file()
]
```
A more generic solution might be to create a `downloads` directory under each of the individual data source directories, and put all of the files we want to make available as supplemental resources in there. Then they'd be clearly identified by their location, and could be of any file type. But if everything is a PDF right now then this seems fine.